src/test/java/org/cloudfoundry/promregator/auth/OAuth2XSUAAEnricherTest.java

```diff
@@ -67,31 +68,77 @@ void testAppropriateJWTCall() {
-		
-		OAuth2XSUAAAuthenticationConfiguration authenticatorConfig = new OAuth2XSUAAAuthenticationConfiguration();
-		authenticatorConfig.setClient_id("client_id");
-		authenticatorConfig.setClient_secret("client_secret");
-		authenticatorConfig.setTokenServiceURL("http://localhost:9001/oauth/token");
-		
-		OAuth2XSUAAEnricher subject = new OAuth2XSUAAEnricher(authenticatorConfig);
-		
+
+		OAuth2XSUAAEnricher subject = new OAuth2XSUAAEnricher(getConfig());
+
 		HttpGet mockGet = Mockito.mock(HttpGet.class);
 		Mockito.when(mockGet.getURI()).thenAnswer(new Answer<URI>() {
 
 			@Override
 			public URI answer(InvocationOnMock invocation) throws Throwable {
 				return new URI("http://localhost/target");
 			}
-			
+
 		});
-		
+
 		// first call will trigger OAuth request
 		subject.enrichWithAuthentication(mockGet);
-		
+
 		// second one should not
 		subject.enrichWithAuthentication(mockGet);
-		
+
 		assertThat(this.ams.getOauthTokenHandler().getCounterCalled()).isEqualTo(1);
 	}
-	
+
+	@Test
+	void testAuthorizationHeaderAddedWhenTokenRetrieved() throws Exception {
+		ClientCredentialsTokenFlow tokenClientMock = Mockito.mock(ClientCredentialsTokenFlow.class);
+		Mockito.when(tokenClientMock.execute())
+				.thenReturn(new OAuth2TokenResponse("someAccessToken", 42l, "someRefreshToken"));
+
+		ArgumentCaptor<String> captor = ArgumentCaptor.forClass(String.class);
+
+		OAuth2XSUAAEnricher subject = new OAuth2XSUAAEnricher(getConfig("aScope anotherScope"), tokenClientMock);
+
+		HttpGet get = new HttpGet();
+
+		assertThat(get.getAllHeaders()).isEmpty();
+
+		subject.enrichWithAuthentication(get);
+
+		Mockito.verify(tokenClientMock).scopes(captor.capture());
+		assertThat(captor.getAllValues()).contains("aScope", "anotherScope");
+		assertThat(captor.getAllValues()).hasSize(2);
+
+		assertThat(get.getHeaders("Authorization")[0].getValue()).isEqualTo("Bearer someAccessToken");
+	}
+
+	@Test
+	void testAuthorizationHeaderNotAddedAndNoExceptionThrownWhenTokenNotRetrieved() throws Exception {
+		ClientCredentialsTokenFlow tokenClientMock = Mockito.mock(ClientCredentialsTokenFlow.class);
+		Mockito.when(tokenClientMock.execute()).thenThrow(new TokenFlowException("ups, something went wrong"));
+
+		OAuth2XSUAAEnricher subject = new OAuth2XSUAAEnricher(getConfig(), tokenClientMock);
+
+		HttpGet get = new HttpGet();
+
+		subject.enrichWithAuthentication(get);
+
+		assertThat(get.getAllHeaders()).isEmpty();
+	}
+
+	private OAuth2XSUAAAuthenticationConfiguration getConfig() {
+		return getConfig(null);
+	}
+	private OAuth2XSUAAAuthenticationConfiguration getConfig(String scopes) {
+		OAuth2XSUAAAuthenticationConfiguration authenticatorConfig = new OAuth2XSUAAAuthenticationConfiguration();
+		authenticatorConfig.setClient_id("client_id");
+		authenticatorConfig.setClient_secret("client_secret");
+		authenticatorConfig.setTokenServiceURL("http://localhost:9001/oauth/token");
+		if (! Strings.isNullOrEmpty(scopes)) {
```

Comment:
minor & matter of taste: I am not so a fan of the google base methods. So far, we have preferred apache commons3. There it would be `StringUtils.isEmpty`.
But as said: that's more a matter of taste...

Reply:
done